### PR TITLE
OOP school library: unit tests

### PR DESCRIPTION
**Pull request by nwakauc**
## In this Milestone we implemented the following ✔;

- Installed `Rspec`;
- Added unit tests for all the methods in all the classes:
- `Book`
- `Person`
- `Student`
- `Teacher`
- `Classroom`
- `Rental`
- **Thank you for reviewing our code 😎.**

### CODE REVIEW

File: spec/book_spec.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  end
+
+  it 'Check for correct author' do
+    expect(@book.author).to eq('Shakir')
+  end
+
+  it 'Check for rental length' do
+    expect(@book.rentals.length).to eq(0)
+  end
+end
```
Comment (Whoistolu) [inline spec/book_spec.rb]: - [x] [REQUIRED] You have done a god job testing the `initialize` method, however, a part of the project requirement stated that you should have unit tests for all the methods in the classes. Kindly add a test for the ` add_rental` method.

File: spec/person_spec.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    expect(name).to eq 'James'
+  end
+  it 'Check  if the add_rental return Rental ' do
+    date = '2023/03/15'
+    book = Book.new('Title', 'Author')
+    person = Person.new(50, 'James')
+    rental = book.add_rental(person, date)
+    expect(rental).to be_instance_of Rental
+  end
+end
```
Comment (Whoistolu) [inline spec/person_spec.rb]: - [x] [REQUIRED] Hello team, it looks like we also forgot to test this private method `of_age?` for the Person class as well. This is a hard requirement so let's fix it 💪🏽

File: spec/student_spec.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  end
+
+  it 'checks for Student Name' do
+    expect(@student.name).to eql('Shakir')
+  end
+
+  it 'checks for Student Age' do
+    expect(@student.age).to eql(16)
+  end
+end
```
Comment (Whoistolu) [inline spec/student_spec.rb]: - [x] [REQUIRED] Great job team, however, in the Student class we need to test for the `add_classroom`, and `play_hooky` methods respectively as stated by the project requirement.

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi Team,

@uchexm and @Shakir-Hussain12 
Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

## N/A

Cheers, and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi Team,

#### Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

### Congratulations! 🎉 

giphy (1)

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**